internal/parse/parse.go

```diff
@@ -99,13 +142,25 @@ func (p *flagParser) parseValue(stopSet string) (interface{}, error) {
 
 	switch in[0] {
 	case '[':
-		return p.parseArray()
+		if p.cfg.Array {
+			return p.parseArray()
+		}
+		return p.parsePrimitive(stopSet)
```

Comment:
Should we adapt the stop set based on the config? e.g. if Array and Object are false and the string does not start with quotes we could return the original string as is.

Reply:
In the case that Array and Object are false the stopSet will just be `","` (the topLevelStopSet) and it will not change unless the parser hits a double or single quoted string (when enabled).

Being that we want top level array parsing to still be allowed when `p.cfg.Array` is false, I believe this path is correct. I am going to add more tests for all the combinations of configs.

Reply:
Oh, I see. Top-level array parsing can not be disabled and is done independent of the `cfg.Array` setting. My impression was that `Array` parsing also disables the top-level array support.

I think that makes sense. Was just not what I expected. Maybe we should document that top-level array splitting is active always.